app/controllers/kudos_controller.rb

```diff
@@ -21,9 +29,26 @@ def create
     end
   end
 
+  def update
+    if kudo.update(kudo_params)
+      redirect_to kudos_path, notice: 'Kudos was successfully updated.'
+    else
+      render :edit, locals: { kudo: kudo }
+    end
+  end
+
+  def destroy
+    kudo.destroy
+    redirect_to kudos_url, notice: 'Kudos was successfully destroyed.'
+  end
+
   private
 
   def kudo_params
     params.require(:kudo).permit(:title, :content, :receiver_id)
   end
+
+  def kudo
+    @kudo ||= Kudo.find(params[:id])
+  end
```

Comment:
Memoization is great tool, but is it really useful here?
Each request generates new instance of the controller, so there is no way the `kudo` will be reused, and you will gain anything from applying this pattern

Reply:
You are right in this case, when a given value is not used often, it does not make sense.
I will try to create a TIL tomorrow

In addition, I noticed that I didn't have alerts made so I added them